### PR TITLE
Generic handling of tree parameters

### PR DESCRIPTION
**Pull request by eileen-kuehn**
This pull request introduces a generic handling of tree parameters and includes

* [x] introduction of generic `ParameterEvent`
* [x] properly handling values for each type of node / event
* [x] only events for supported events are generated to reduce processing time

### CODE REVIEW

File: assess/generators/event_generator.py [hunk 1]
```diff
@@ -1,6 +1,19 @@
+from assess.events.events import ProcessStartEvent, ProcessExitEvent, TrafficEvent
+
+
 class EventGenerator(object):
-    def __init__(self, streamer=None):
+    def __init__(self, streamer=None, supported=None):
         self._streamer = streamer
+        if supported is None:
+            if hasattr(streamer, "supported"):
```
Comment (maxfischer2781) [inline assess/generators/event_generator.py]: You can use ``getattr`` for get-or-default in one go
```suggestion
            supported = getattr(streamer, "supported", {ProcessStartEvent: True, ...}):
```

**Review by maxfischer2781 — CHANGES_REQUESTED** [review]
See comment for possible improvement. Otherwise looks good to go.

File: assess/generators/event_generator.py (hunk 1 above)
Comment (eileen-kuehn) [reply assess/generators/event_generator.py]: Thanks, I included this.

**Review by maxfischer2781 — APPROVED** [review]

**Review by maxfischer2781 — APPROVED** [review]
👍

**Pull request merged**